File: CPP/TileBitmapVisual.cpp
/*
========================================================================

	A visual the size of a tile, so is rendered in tilespace

    Author: Tim Wheeler
    Contact: timwheeleronline@gmail.com

========================================================================
*/

#include "TileBitmapVisual.h"

using namespace MysticDave;

TileBitmapVisual::TileBitmapVisual( Pos2D * pos, ALLEGRO_BITMAP * bitmap ) : BitmapVisual( pos, bitmap ) {
	// do nothing
}

TileBitmapVisual::~TileBitmapVisual() {
	BitmapVisual::~BitmapVisual();
}

void TileBitmapVisual::Render( int x, int y ) {
	al_draw_bitmap( bitmap, pos->GetPixX() - x, pos->GetPixY() - y, 0 );
}

void TileBitmapVisual::Render( int x, int y, float cx, float cy, float cz ) {
	al_draw_tinted_bitmap( bitmap, al_map_rgba_f(cx, cy, cz, 1), pos->GetPixX() - x, pos->GetPixY() - y, 0 );
}

File: Java/Visuals/TileBitmapVisual.java
/*
========================================================================

	A visual the size of a tile, so is rendered in tilespace

========================================================================
*/
package Visuals;

import MysticDave.Chamber;
import MysticDave.Vec2i;
import org.newdawn.slick.Graphics;
import org.newdawn.slick.Image;

public class TileBitmapVisual extends BitmapVisual {
    
    public TileBitmapVisual( Vec2i pos, Image bitmap ) {
        super( pos, bitmap );
    }
    
    @Override
    public void Render( Graphics g ) {
        g.drawImage( bitmap, pos.x*Chamber.TILE_PIXEL_DIM, pos.y*Chamber.TILE_PIXEL_DIM );
    }
    
}
